daffodil-core/src/main/scala/org/apache/daffodil/grammar/AlignedMixin.scala

```diff
@@ -125,8 +126,8 @@ trait AlignedMixin extends GrammarMixin { self: Term =>
   }
 
   private lazy val priorAlignmentApprox: AlignmentMultipleOf = {
-    if (this.isInstanceOf[Root]) {
-      AlignmentMultipleOf(0) // root is aligned with anything
+    if (this.isInstanceOf[Root] || this.isInstanceOf[DetachedElementDecl]) {
+      AlignmentMultipleOf(0) // root and detached elements are aligned with anything
```

Comment:
Don't understand comment here. Why are detached elements always aligned?

Reply:
As I read the spec, alignment doesn't really apply to prefix lengths. When parsing an element that has prefix length, alignment bits are skipped if necessary based on the alignment of the element. Then the prefix length is immediately parsed, then the element is immediately parsed. To ensure no alignment is done, we just pretened that the prefixed element is aligned with everything.

Reply:
So the use case that needs to work right is just a 32-bit word-aligned prefix-length integer which gives the prefix length of a "normally thought of as" byte-aligned text string. The element itself shouldn't have to specify 32-bit alignment. That should happen because the prefix is a 32-bit aligned integer word.  I believe this is actually a PL/1 data situation. Strings all have prefixes. Those are always aligned words (either 4 bytes or 2 bytes I believe). Let's just assume 4 bytes for the moment. Then two 7-character strings would look like for the first string, 4-byte length, 7-bytes of text,  then for the 2nd string 1 byte of alignment,  4-byte length, 7-bytes of text.

Reply:
I'll add a test to make sure that works (I think it should). The spec does say:

> It is a schema definition error if the xs:simpleType [referenced by prefixLengthType] specifies any of:
>
> ...
> dfdl:alignment other than '1
> ...

So I think the alignment/Units properties needs to be put on the element, but the skipped bits should occur before the prefix length element is parsed.

Reply:
I think some of this stuff is wrong in the spec. But let's implement what it says there, and then consider correcting it.

I believe none of the properties on an element having lengthKind 'prefixed' should affect the representation of the prefix itself.  Properties like prefixIncludesPrefixLength affect the interpretation of the value from that prefix, but not the representation of the prefix. 

I believe all  properties that are supposed to apply to the prefix itself must be expressed on a simple type definition. There should be no property combining between the prefix's simple type, i.e., the pseudo-element you create for it, and the element having that prefix type. 

Let's take the case of a string with a 4-byte binary integer as prefix length. The integer has to be 32-bit aligned.

Properties on the element want to be those associated with a textual string, not those for a binary integer. 

Consider a harder case: Consider the prefix length is itself a text integer, giving the length of another text integer. Then if the property dfdl:textNumberJustification and dfdl:textTrimKind, etc. appear on the element, do those apply to the element, the prefix or both. I claim they apply ONLY to the element, NOT to the prefix. Any properties for the prefix have to be expressed on the simple type definition, which will have to be a named type definition someplace if any properties are needed.

Reply:
I don't think there is any property combining going on. Some properties just aren't allowed to be on the prefix type (e.g. alignment,separator, initiators, leading/trailing skip). These all seem related to framing which kindof makes sense since the prefix type is withing the framing of the element. So any framing must comes from the element, and that surrounds both the prefix length and the element. If framing properties were allowed on the prefix element then you would have framing for the prefix inside framing of the elemtn, which seems a bit odd. I imagine this was just to simplify things--prefix lenghts are already kindof complicated.

For the example textNumberJustification/textTrimKind, that works as one would expected (I have tests for this). If those properties are on the the element then it affects the parsing of the element, if they are on the simpleType then it affects the parsing of the prefix length.